src/loop.cpp

```diff
@@ -29,6 +29,111 @@ namespace oxen::quic
         });
     }
 
+    /** Static casting to `decltype(timeval::tv_{sec,usec})` makes sure that;
+        - on linux
+            .tv_sec is type __time_t
+            .tv_usec is type __suseconds_t
+        - on OSX    (https://developer.apple.com/documentation/kernel/timeval)
+            .tv_sec is type __darwin_time_t
+                - this is an annoying typedef of `time_t`
+            .tv_usec is type __darwin_suseconds_t
+                - this is an equally annoying typedef for `suseconds_t`
+        Alas, yet again another mac idiosyncrasy...
+     */
+    timeval loop_time_to_timeval(loop_time t)
+    {
+        return timeval{
+                .tv_sec = static_cast<decltype(timeval::tv_sec)>(t / 1s),
+                .tv_usec = static_cast<decltype(timeval::tv_usec)>((t % 1s) / 1us)};
+    }
+
+    bool EventHandler::start()
+    {
+        if (_is_running or _is_stopped)
+            return false;
+
+        if (event_add(ev.get(), &interval) != 0)
+        {
+            log::critical(log_cat, "EventHandler failed to start repeating event!");
+            return false;
+        }
+
+        _is_running = true;
+
+        return true;
+    }
+
+    bool EventHandler::pause()
+    {
+        if (not _is_running or _is_stopped)
+            return false;
+
+        if (event_del(ev.get()) != 0)
+        {
+            log::critical(log_cat, "EventHandler failed to pause repeating event!");
+            return false;
+        }
+
+        _is_running = false;
+
+        return true;
+    }
+
+    bool EventHandler::stop()
+    {
+        if (not _is_running and _is_stopped)
+        {
+            log::critical(log_cat, "EventHandler is already permanently stopped!");
+            return false;
+        }
+
+        ev.reset();
+        f = nullptr;
```

Comment:
This doesn't seem thread-safe: someone could call `stop()` from any thread which means the clearing of the existing `f` could happen while `f` is currently in the middle of a call in the event loop.

(This becomes irrelevant if we get rid of `stop()` in favour of using the destructor, which *is* thread-safe because destruction gets deferred to the event loop).